driver/src/main/scala/com/stratio/sparkta/driver/SparktaJob.scala

```diff
@@ -83,39 +85,37 @@ object SparktaJob extends SLF4JLogging {
   @tailrec
   def applyParsers(input: DStream[Event], parsers: Seq[Parser]): DStream[Event] = {
     parsers.headOption match {
-      case Some(headParser: Parser) => applyParsers(input.map(event => {
-        Try(headParser.parse(event)) match {
-          case Success(okEvent) => Some(okEvent)
-          case Failure(exception) => {
-            val error = s"Failure[Parser]: ${event.toString} | Message: ${exception.getLocalizedMessage}" +
-              s" | Parser: ${headParser.getClass.getSimpleName}"
-            log.error(error, exception)
-            None
-          }
-        }
-      }).flatMap(event => event match {
-        case Some(value) => Seq(value)
-        case None => Seq()
-      }), parsers.drop(1))
+      case Some(headParser: Parser) =>
+        applyParsers(input.map(event =>parseEvent(headParser, event))
+        .flatMap(eventToSeq(_)), parsers.drop(1))
```

Comment:
This is equivalent to .flatMap(eventToSeq). Maybe you prefer to omit parentheses, or put the complete statement: .flatMap(event => eventToSeq(event)). I think the first one is better option: higher order functions as argument.